src/main/kotlin/org/elaastic/questions/player/components/dashboard/DashboardModelFactory.kt

```diff
@@ -0,0 +1,232 @@
+            LearnerStateOnPhase.WAITING
+        } else if (hasResponseForPhase) {
+            LearnerStateOnPhase.ACTIVITY_TERMINATED
+        } else {
+            LearnerStateOnPhase.ACTIVITY_NOT_TERMINATED
+        }
+    }
+
+    /**
+     * Get the LearnerStateOnPhase of the attendee on the Evaluation phase.
+     *
+     * A learner has finished the phase if he has evaluated exactly the number
+     * of responses required.
+     *
+     * @param attendee the attendee
+     * @param sequence the sequence
+     * @param evaluationPhaseState the state of the evaluation phase
+     * @param evaluationCountByUser the number of evaluations made by each user
+     * @return the LearnerStateOnPhase base on the argument
+     * @see EvaluationSpecification
+     */
+    private fun getAttendeeStateOnEvaluationPhase(
+        attendee: LearnerAssignment,
+        sequence: Sequence,
+        evaluationPhaseState: DashboardPhaseState,
+        evaluationCountByUser: Map<LearnerAssignment, Long>
+    ): LearnerStateOnPhase {
+
+        // To not use any Service in a Factory, we need to pass the evaluationCountByUser as an argument
+        val nbEvaluationMade = evaluationCountByUser[attendee]
+
+        val hasMadeAllEvaluationForPhase: Boolean =
+            try {
+                sequence.getEvaluationSpecification().responseToEvaluateCount.toLong() == nbEvaluationMade
+            } catch (e: IllegalStateException) {
+                false /* If the sequence isn't initialized an Exception his throw by the getEvaluationSpecification function */
+            }
+
+        return if (evaluationPhaseState == DashboardPhaseState.NOT_STARTED) {
+            LearnerStateOnPhase.WAITING
+        } else if (hasMadeAllEvaluationForPhase) {
+            LearnerStateOnPhase.ACTIVITY_TERMINATED
+        } else {
+            LearnerStateOnPhase.ACTIVITY_NOT_TERMINATED
+        }
+    }
+
+    /**
+     * Get the LearnerStateOnPhase of the attendee on the Read phase. (eq. the
+     * result phase)
+     *
+     * By default, the learner has not terminated the activity
+     *
+     * If the sequence is *Face-to-Face or blended and the read phase is in
+     * progress, the learner is waiting.
+     *
+     * If the sequence is *Distance*, we need to check if the learner has
+     * terminated the previous activity, the Evaluation phase. For that we
+     * need to get the argument for the [getAttendeeStateOnEvaluationPhase]
+     * function.
+     *
+     * @param sequence the sequence
+     * @param readPhaseState the state of the read phase
+     * @param attendee the attendee
+     * @param evaluationPhaseState the state of the evaluation phase
+     * @param evaluationCountByUser the number of evaluations made by each user
+     * @return the state of the attendee on the read phase
+     * @see LearnerStateOnPhase
+     * @see getAttendeeStateOnEvaluationPhase
+     */
+    private fun getAttendeeStateOnReadPhase(
```

Comment:
Do we still need all that stuff since we don't display any information on the third phase?

Reply:
fix in ffa31d94fe688bb4bb7e2cea90fa2f0a91a4ae37